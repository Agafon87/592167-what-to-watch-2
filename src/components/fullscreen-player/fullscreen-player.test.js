import React from "react";
import renderer from "react-test-renderer";
import FullscreenPlayer from "./fullscreen-player.jsx";

const mock = {
  currentTime: 1,
  duration: 1,
  isLoading: true,
  renderPlayer: jest.fn(),
  isPlaying: true,
  isFullScreen: true,
  onSwitchPlayer: jest.fn(),
  onTogglePlay: jest.fn(),
  onToggleFullScreen: jest.fn(),
  onSetFullScreen: jest.fn(),
  onResetPlayer: jest.fn(),
};

describe(`FullscreenPlayer component`, () => {
  const {
    currentTime,
    duration,
    isLoading,
    renderPlayer,
    isPlaying,
    isFullScreen,
    onSwitchPlayer,
    onTogglePlay,
    onToggleFullScreen,
    onSetFullScreen,
    onResetPlayer
  } = mock;

  it(`renders correctly`, () => {
    const tree = renderer.create(<FullscreenPlayer
      currentTime={currentTime}
      duration={duration}
      isLoading={isLoading}
      isPlaying={isPlaying}
      isFullScreen={isFullScreen}
      onResetPlayer={onResetPlayer}
      onTogglePlay={onTogglePlay}
      onToggleFullScreen={onToggleFullScreen}
      onSwitchPlayer={onSwitchPlayer}
      onSetFullScreen={onSetFullScreen}
      renderPlayer={renderPlayer}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
