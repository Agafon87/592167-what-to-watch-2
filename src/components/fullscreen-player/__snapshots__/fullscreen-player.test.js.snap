// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullscreenPlayer component renders correctly 1`] = `
<div
  className="fullscreen fullscreen-enabled"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="player"
  >
    <div
      className="player__preloader"
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
        }
      }
    >
      <img
        alt="preload"
        height="50"
        src="/img/preloader.gif"
        width="50"
      />
    </div>
    <button
      className="player__exit"
      onClick={[MockFunction]}
      type="button"
    >
      Exit
    </button>
    <div
      className="player__controls"
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max={1}
            value={1}
          />
          <div
            className="player__toggler"
            style={
              Object {
                "left": "100%",
              }
            }
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          00:00:00
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[MockFunction]}
          type="button"
        >
          <svg
            height="21"
            viewBox="0 0 14 21"
            width="14"
          >
            <use
              xlinkHref="#pause"
            />
          </svg>
          <span>
            Pause
          </span>
        </button>
        <div
          className="player__name"
        >
          Transpotting
        </div>
        <button
          className="player__full-screen"
          onClick={[MockFunction]}
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
